packages/comms/src/services/wsWorkunits.ts

```diff
@@ -10,6 +10,49 @@ import { ESPConnection, Service } from "../espConnection";
     * http://json2ts.com/
 */
 
+export {
+    WsWorkunits
+};
+
+export namespace WsWorkunitsEx {
+
+    export interface WUPublishWorkunit {
+        Wuid: string;
+        Cluster: string;
+        JobName: string;
+        Activate: boolean;
+        QueryName?: string;
+        WorkUnitJobName?: string;
+        NotifyCluster?: boolean;
+        Wait?: number;
+        NoReload?: boolean;
+        UpdateWorkUnitName?: boolean;
+        memoryLimit?: string;
+        TimeLimit?: number;
+        WarnTimeLimit?: number;
+        Priority?: string;
+        RemoteDali?: string;
+        Comment?: string;
+        DontCopyFiles?: boolean;
+        SourceProcess?: string;
+        AllowForeignFiles?: boolean;
+        UpdateDfs?: boolean;
+        UpdateSuperFiles?: boolean;
+        UpdateCloneFrom?: boolean;
+        AppendCluster?: boolean;
+        IncludeFileErrors?: boolean;
+        DfuCopyFiles?: boolean;
+        DfuQueue?: string;
+        DfuWait?: number;
+        DfuOverwrite?: boolean;
+        OnlyCopyFiles?: boolean;
+        StopIfFilesCopied?: boolean;
+        DfuPublisherWuid?: string;
+        RemoteStorage?: string;
+    }
```

Comment:
Where did this come from?

Reply:
I'd added that manually for use as the type in the extended publish function. The interfaces generated from the WSDL don't currently respect mandatory/optional parameters. Or rather, it often seems to imply every property is optional, which doesn't seem correct. Unless I'm misunderstanding what minOccurs=0 means. I was going to ask about that in the next ECL Watch meeting. 

![image](https://github.com/hpcc-systems/Visualization/assets/79224539/871aa119-e213-40f3-92b7-e36aa18a4db5)